tox.ini

```diff
@@ -72,37 +70,47 @@ commands=
     # due to multiple Trinity instances competing for the same ports
     py.test -n 1 {posargs:tests/trinity/integration/}
 
+[testenv:py36-trinity-integration]
+deps = {[common-trinity-integration]deps}
+passenv = {[common-trinity-integration]passenv}
+commands = {[common-trinity-integration]commands}
```

Comment:
@veox Nice! Didn't know that syntax was possible. Am I right to assume that we are now entirely relying on the Python version that is available on the system (e.g. just taking the one that is defined in the circle image)? So, if one runs these `tox` jobs locally `testenv:py36-trinity-integration` and `testenv:py37-trinity-integration` will not use different Python versions. If so, that would be a small regression. Not sure how important that is for anyone. I'm also not sure if there's maybe another command line argument one could pass to `tox` to specify the python version for each job anyway. I'm a bit in a hurry so not looking it up.

Reply:
As I understand, specifying `basepython` in `testenv:py36-trinity-integration` was overriding what was specified in `testenv`. Effectively, `testenv` had (and [still has](https://github.com/veox/py-evm/blob/27ebdf83481540098413de90af54d98077a7e8ed/tox.ini#L46)) instructions to use `basepython=python3.6` for all environments/tests that have `py36` in their name (such as `testenv:py36-trinity-integration`).

That's my understanding - but I could be wrong. Need to check...

Reply:
In [a different branch](https://github.com/veox/py-evm/commits/run-trinity-integration-tests-with-py3.7-sloppy-check), I've set `basepython` to `false` for `py36` jobs (see commit https://github.com/veox/py-evm/commit/5de2944f16b361a12ec11b1882ce17591dc30536).

All CircleCI jobs that were running `tox` `testenv`s with `py36` in their name failed (e.g. [this one](https://circleci.com/gh/veox/py-evm/846)). They are certainly not using the `python` available on the system, but do indeed pick up the one specified in the "generic" `testenv`.

Reply:
To clarify, the particular line you're commenting on, with curly syntax like `{[common-trinity-integration]deps}`, is a different configuration feature, [substitutions](https://tox.readthedocs.io/en/latest/config.html#substitution-for-values-from-other-sections). It instructs to use the value from a different section, `common-trinity-integration`.